packages/ra-core/src/form/Form.spec.tsx

```diff
@@ -167,6 +167,33 @@ describe('Form', () => {
         });
     });
 
+    it('should not display a notification on submit when invalid', async () => {
+        const Notification = () => {
+            const { notifications } = useNotificationContext();
+            return notifications.length > 0 ? (
+                <div>{notifications[0].message}</div>
+            ) : null;
+        };
+
+        render(
+            <CoreAdminContext>
+                <>
+                    <Form onSubmit={jest.fn()} disableInvalidFormNotification>
+                        <Input source="name" validate={required()} />
+                        <button type="submit">Submit</button>
+                    </Form>
+                    <Notification />
+                </>
+            </CoreAdminContext>
+        );
+
+        fireEvent.click(screen.getByText('Submit'));
+        await waitFor(() => {
+            const notification = screen.queryByText('ra.message.invalid_form');
+            expect(notification).toBeNull();
+        });
```

Comment:
This test passes even without `disableInvalidFormNotification`. 

It is not always easy to check for the absence of something with Jest/RTL. You need to find an event to await for, that proves the form submission is complete, before attempting to check if the notification is there or not. You can try providing a `jest.fn()` as `onSuccess` to see if it helps, or try rendering the record and have RTL look for the changed value in the record.

Reply:
Thanks for feedback! 🙏 
I updated the unit test so that I'm waiting for the validation message to be shown before checking the absence of the notification.